src/odemis/gui/cont/acquisition.py

```diff
@@ -927,6 +928,203 @@ def on_file_export_done(self, future):
+        self._tab_panel = tab_panel
+        self._projectbar_ctrl = projectbar_ctrl
+        self._calibration_ctrl = calibration_ctrl
+
+        # Path to the acquisition
+        # <External storage>/<date>/<project name>/<roa name>
+        # TODO: display proper path to external storage (if mounted)
+        self.path = datetime.today().strftime('%Y-%m-%d')
+        self._tab_panel.txt_destination.SetValue("<External Storage>/%s" % self.path)
+
+        # ROA count
+        self.roa_count = 0
+        self._tab_panel.txt_num_rois.SetValue("0")
+
+        # For acquisition
+        self.btn_acquire = self._tab_panel.btn_sparc_acquire
+        self.btn_cancel = self._tab_panel.btn_sparc_cancel
+        self.gauge_acq = self._tab_panel.gauge_sparc_acq
+        self.lbl_acqestimate = self._tab_panel.lbl_sparc_acq_estimate
+        self.txt_num_rois = self._tab_panel.txt_num_rois
+        self.bmp_acq_status_warn = self._tab_panel.bmp_acq_status_warn
+        self.bmp_acq_status_info = self._tab_panel.bmp_acq_status_info
+
+        # Link buttons
+        self.btn_acquire.Bind(wx.EVT_BUTTON, self.on_acquisition)
+        self.btn_cancel.Bind(wx.EVT_BUTTON, self.on_cancel)
+
+        # Hide gauge, disable acquisition button
+        self.gauge_acq.Hide()
+        self._tab_panel.Parent.Layout()
+        self.btn_acquire.Enable(False)
+
+        # Update text controls when projects/roas/rocs are changed
+        self.roa_subscribers = []  # list of ROA subscribers (to make sure we don't subscribe to the same ROA twice)
+        tab_data.projects.subscribe(self._on_projects, init=True)
+        for roc in self._tab_data_model.calibration_regions.value.values():
+            roc.coordinates.subscribe(self._on_roc)
+
+    def _on_projects(self, projects):
+        for p in projects:
+            p.roas.subscribe(self._on_roas)
+
+    def _on_roas(self, roas):
+        # For each roa, subscribe to calibration attribute. Make sure to update acquire button / text if ROC is changed.
+        for roa in roas:
+            if roa not in self.roa_subscribers:
+                roa.roc.subscribe(self._on_roc)
+                self.roa_subscribers.append(roa)
+        self.update_roa_count()
+        self.check_acquire_button()
+        self.update_acquisition_time()  # to update the message
+
+    def _on_roc(self, _):
+        self.check_acquire_button()
+        self.update_acquisition_time()  # to update the message
+
+    def check_acquire_button(self):
+        self.btn_acquire.Enable(self.roa_count and not self._get_undefined_calibrations())
+
+    @wxlimit_invocation(1)  # max 1/s
+    def update_acquisition_time(self):
+        lvl = None  # icon status shown
+        if self.roa_count == 0:
+            lvl = logging.WARN
+            txt = "No region of interest selected."
+        elif self._get_undefined_calibrations():
+            lvl = logging.WARN
+            txt = "Calibration regions %s missing." % (self._get_undefined_calibrations(),)
```

Comment:
That will look like "Calibration regions [2, 7] missing.", right? It's a little ugly. You could either use `", ".join()`, or just indicate the number of regions missing?